Gems/AWSCore/Code/Source/Editor/UI/AWSCoreResourceMappingToolAction.cpp

```diff
@@ -6,133 +6,95 @@
  *
  */
 
-#include <AzCore/IO/FileIO.h>
 #include <AzCore/Utils/Utils.h>
-#include <AzFramework/StringFunc/StringFunc.h>
 
 #include <AWSCoreInternalBus.h>
+#include <Configuration/AWSCoreConfiguration.h>
 #include <Editor/UI/AWSCoreResourceMappingToolAction.h>
 
 namespace AWSCore
 {
-    AWSCoreResourceMappingToolAction::AWSCoreResourceMappingToolAction(const QString& text)
-        : QAction(text)
+    AWSCoreResourceMappingToolAction::AWSCoreResourceMappingToolAction(const QString& text, QObject* parent)
+        : QAction(text, parent)
         , m_isDebug(false)
         , m_enginePythonEntryPath("")
         , m_toolScriptPath("")
         , m_toolQtBinDirectoryPath("")
-        , m_toolLogPath("")
+        , m_toolLogDirectoryPath("")
+        , m_toolConfigDirectoryPath("")
         , m_toolReadMePath("")
```

Comment:
FYI initializing string classes with `""` is a pessimization.
It is actually requires strlen to be called on the supplied char* to determine that the string is empty.
It is better to let the type be defaulted constructed and not add it to the member initializer list here.